src/app/components/ThemeProvider/index.tsx

```diff
@@ -1,34 +1,76 @@
 import React, { PropsWithChildren } from 'react';
+import type { LoadableComponent } from '@loadable/component';
+import nodeLogger from '#lib/logger.node';
+import { THEME_PROVIDER_ERROR } from '#app/lib/logger.const';
 import defaultServiceVariants from '../../lib/config/services/defaultServiceVariants';
-import { Services, Variants } from '../../models/types/global';
+import {
+  ServicesVariantsProps,
+  Variants,
+  ServicesWithVariants,
+  ServicesWithNoVariants,
+} from '../../models/types/global';
 import themes from './themes/loadableConfig';
 import fallBackTheme from './themes/news';
 
-interface Props {
-  service: Services;
-  variant?: Variants | null;
-}
+const logger = nodeLogger(__filename);
+
+type ThemeComponentLoadable = LoadableComponent<{ children: React.ReactNode }>;
+type FallbackThemeComponent = React.FC<{ children: React.ReactNode }>;
+
+type NonVariantThemesType = {
+  [_service in ServicesWithNoVariants['service']]: ThemeComponentLoadable;
+};
+
+type VariantThemesType = {
+  [_service in ServicesWithVariants['service']]: {
+    [_variant in Variants]?: ThemeComponentLoadable;
+  };
+};
+
+const nonVariantThemes = Object.fromEntries(
+  Object.entries(themes).filter(([_service, theme]) =>
+    Object.keys(theme).includes('render'),
+  ),
+) as unknown as NonVariantThemesType;
+
+const variantThemes = Object.fromEntries(
+  Object.entries(themes).filter(
+    ([_service, theme]) => !Object.keys(theme).includes('render'),
+  ),
+) as unknown as VariantThemesType;
 
 export const ThemeProvider = ({
   children,
   service,
-  variant,
-}: PropsWithChildren<Props>) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let LoadableContextProvider: any = fallBackTheme;
+  ...rest
+}: PropsWithChildren<ServicesVariantsProps>) => {
+  let LoadableContextProvider:
+    | ThemeComponentLoadable
+    | FallbackThemeComponent
+    | null = null;
 
-  const serviceVariant: Variants = variant || defaultServiceVariants[service];
+  const variant: Variants = rest.variant || defaultServiceVariants[service];
 
-  if (serviceVariant === 'default' || !serviceVariant) {
-    LoadableContextProvider = themes[service];
+  let serviceVariants;
+  let serviceNoVariants;
+
+  if (service in variantThemes) {
+    serviceVariants =
```

Comment:
```suggestion
    serviceVariantTheme =
```
I found it hard to follow this was the resolved theme, just suggesting some different naming